version: '3'
services:
  reverse:
    image: traefik:v2.0
    ports:
      - "443:443"
      - "8080:8080"
      - "80:80"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./letsencrypt:/letsencrypt
    command:
      - "--api=true"
      - "--api.dashboard=true"
      - --entrypoints.dashboard.address=:80
      - --providers.docker
      - --api.insecure # Don't do that in production 
      - "--api.debug=true"
      - "--providers.docker=true"
      - --entrypoints.websecure.address=:443
     # - "--entrypoints.dashboard.address=:8080"
      - "--certificatesresolvers.mydnschallenge.acme.dnschallenge=true"
      - "--certificatesresolvers.mydnschallenge.acme.dnschallenge.provider=cloudflare"
      - "--certificatesresolvers.mydnschallenge.acme.email=malek.attia@telecom-sudparis.eu"
      - "--certificatesresolvers.mydnschallenge.acme.storage=/letsencrypt/acme.json"      
      - "--log.level=DEBUG"
    environment:
      CF_API_EMAIL: $CF_API_EMAIL
      CF_API_KEY: $CF_API_KEY
    labels:
      - "traefik.http.routers.api.rule=Host(`pfechercheriattia.fr`)  && (PathPrefix(`/api`) || PathPrefix(`/dashboard`))"
      - "traefik.http.routers.api.entrypoints=dashboard"
      - "traefik.http.routers.api.service=api@internal"
  cells:
    image: pydio/cells:latest
    ports:
      - "8888:80"
    environment:
      - CELLS_BIND=https://0.0.0.0:80
      - CELLS_EXTERNAL=https://pydio.pfechercheriattia.fr
      - CELLS_NO_TLS=1
    labels:
      #- "traefik.enable=true"
      #- "traefik.http.routers.cells.rule=Host(`pydio.pfechercheriattia.fr`)"
      #- "traefik.http.routers.cells.entrypoints=websecure"
      #- "traefik.http.routers.cells.tls.certresolver=mydnschallenge"
      #- "traefik.http.routers.cells.service=cells"
      #- "traefik.http.services.cells.loadbalancer.server.port=8888"
      #- "traefik.http.routers.cells=cells"
      #- "traefik.http.services.cells.loadbalancer.server.port=80"
       - "traefik.enable=true"
       - "traefik.http.routers.cells.entrypoints=dashboard"
       - "traefik.http.routers.cells.rule=Host(`pydio.pfechercheriattia.fr`)"
       - "traefik.http.middlewares.cells-https-redirect.redirectscheme.scheme=https"
       - "traefik.http.routers.cells.middlewares=cells-https-redirect"
       - "traefik.http.routers.cells-secure.entrypoints=websecure"
       - "traefik.http.routers.cells-secure.rule=Host(`pydio.pfechercheriattia.fr`)"
       - "traefik.http.routers.cells-secure.tls=true"
       - "traefik.http.routers.cells-secure.tls.certresolver=mydnschallenge"
       - "traefik.http.routers.cells-secure.service=cells"
       - "traefik.http.services.cells.loadbalancer.server.port=80"
  mysql:
    image: mysql:5.7
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: P@ssw0rd
      MYSQL_DATABASE: cells
      MYSQL_USER: pydio
      MYSQL_PASSWORD: P@ssw0rd
    command: [mysqld, --character-set-server=utf8mb4, --collation-server=utf8mb4_unicode_ci]
    ports: ["3306:3306"]
